src/prefect/server/api/concurrency_limits.py

```diff
@@ -127,14 +298,53 @@ async def reset_concurrency_limit_by_tag(
     ),
     db: PrefectDBInterface = Depends(provide_database_interface),
 ) -> None:
+    # Reset V2 leases if V2 limit exists; otherwise fall back to V1
+    v2_name = f"tag:{tag}"
+
     async with db.session_context(begin_transaction=True) as session:
-        model = await models.concurrency_limits.reset_concurrency_limit_by_tag(
+        model = await cl_v2_models.read_concurrency_limit(session=session, name=v2_name)
+
+        if not model:
+            # Fall back to V1
+            model = await models.concurrency_limits.reset_concurrency_limit_by_tag(
+                session=session, tag=tag, slot_override=slot_override
+            )
+            if not model:
+                raise HTTPException(
+                    status_code=status.HTTP_404_NOT_FOUND,
+                    detail="Concurrency limit not found",
+                )
+            return
```

Comment:
This logic feels wacky too. I think it'd be easier to read if this was handled after all of the v2 logic.